src/ui/AttachResult/AttachResult.ts

```diff
@@ -0,0 +1,180 @@
+import {
+    Component,
+    ComponentOptions,
+    IComponentBindings,
+    $$,
+    IQueryResult,
+    Initialization,
+    IAnalyticsCaseAttachMeta,
+    IAnalyticsCaseDetachMeta,
+    analyticsActionCauseList
+} from "coveo-search-ui";
+import * as PaperclipIcon from "../../icons/paperclip.svg";
+
+export interface IAttachResultOptions {
+    /** Specifies the tooltip displayed when the result is not attached. */
+    attachCaption?: string;
+    /** Specifies the tooltip displayed when the result is already attached. */
+    detachCaption?: string;
+    /** The function that is called when the user wants to attach a result. */
+    attach: (queryResult: IQueryResult) => Promise<void>;
+    /** The function called when the user wants to un-link a result. */
+    detach: (queryResult: IQueryResult) => Promise<void>;
+    /** Optional function to check the initial state of the component. */
+    isAttached?: (queryResult: IQueryResult) => Promise<boolean>;
+}
+
+/**
+ * The AttachResult component allows a user to link a search result to something else in their external
+ * system, for instance a case, incident, request, etc.
+ */
+export class AttachResult extends Component {
+    static ID = "AttachResult";
+
+    protected loading: boolean;
+    protected attached: boolean;
+    protected resultUriHash: string;
+    protected buttonElement: HTMLElement;
+    protected tooltipElement: HTMLElement;
+
+    static options: IAttachResultOptions = {
+        attachCaption: ComponentOptions.buildStringOption({ defaultValue: "Attach Result" }),
+        detachCaption: ComponentOptions.buildStringOption({ defaultValue: "Detach Result" }),
+        attach: ComponentOptions.buildCustomOption(name => () => new Promise<void>((resolve, reject) => { resolve(); }), {
+            defaultFunction: () => () => new Promise<void>((resolve, reject) => { resolve(); })
+        }),
+        detach: ComponentOptions.buildCustomOption(name => () => new Promise<void>((resolve, reject) => { resolve(); }), {
+            defaultFunction: () => () => new Promise<void>((resolve, reject) => { resolve(); })
+        })
+    };
+
+    constructor(
+        public element: HTMLElement,
+        public options: IAttachResultOptions,
+        public bindings?: IComponentBindings,
+        public queryResult?: IQueryResult
```

Comment:
Does queryResult need to be a separate constructor arg, or does it make sense to put it under `options`?

Reply:
What do you usually do with your components that require a result?

Reply:
What i've seen is normally it's in the constructor but i'm not 100% why

Reply:
I see the `ChatterLikedBy` component in the search-ui accepts the `IQueryResult` as a fourth constructor param. What you did is consistent 🤷‍♂️